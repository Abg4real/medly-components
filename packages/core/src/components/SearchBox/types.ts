import { SearchBoxSizes } from '@medly-components/theme';
import { HTMLProps, WithThemeProp } from '@medly-components/utils';

export type Option = {
    label: string;
    value: any;
};

export interface Props extends Omit<HTMLProps<HTMLInputElement>, 'boxSize'>, WithThemeProp {
    /** Size for search box, can be 'S' | 'M' */
    searchBoxSize?: SearchBoxSizes;
    /** Option for search results, in form of label and value */
    options?: Option[];
    /*** maximum height for search results container */
    maxHeight?: string;
    /*** callback to be trigger on selecting the option */
    onOptionSelected?: (value: any) => void;

    getOptions?: () => void;
}
