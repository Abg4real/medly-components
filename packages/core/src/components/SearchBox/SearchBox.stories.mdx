import { useState, useEffect } from 'react';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import { optionsArray, boxSize } from './SearchBox.stories.tsx';
import * as stories from './SearchBox.stories';
import { select } from '@storybook/addon-knobs';
import { SearchBox } from './SearchBox';

<Meta title="In-review" component={SearchBox} />

# SearchBox

### Add a auto suggest text box

<Preview withToolbar>
    <Story name="Default SearchBox">
        <SearchBox
            options={optionsArray}
            searchBoxSize={select('boxSize', boxSize, 'M')}
            onSearchInputChange={value => console.log(value)}
        />
    </Story>
</Preview>

<Preview withToolbar>
    <Story name="SearchBox With All Handlers">
        {() => {
            const [options, setOptions] = useState([]);
            const onSearchInputChangeHandler = value => {
                if (value.length > 2) {
                    setOptions([...optionsArray, { label: 'Test', value: 'Test' }]);
                } else {
                    setOptions([]);
                }
            };
            const onOptionSelectedHandler = option => {
                console.log('option', option);
            };
            return (
                <SearchBox
                    placeholder="Search"
                    onSearchInputChange={onSearchInputChangeHandler}
                    onOptionSelected={onOptionSelectedHandler}
                    options={options}
                    searchBoxSize={select('boxSize', boxSize, 'M')}
                />
            );
        }}
    </Story>
</Preview>

### Props

<Props of={SearchBox} />

##### Small search text box

<Preview>
    <SearchBox searchBoxSize="S" />
</Preview>

##### Medium search text box

<Preview>
    <SearchBox searchBoxSize="M" />
</Preview>

#### Pending things in story

-   [ ] onFocus and onBlur
-   [ ] rgba fix
-   [ ] refactor
-   [x] maxHeight
-   [x] unit tests
-   [x] deploy in storybook for feedback
-   [x] onOptionSelected
-   [x] keyboard selection
-   [x] UI Styles
-   [x] re-add useMemo
-   [x] remove any
-   [x] boxShadow
-   [x] introduce size in storybook

#### Open question

-   should we handle debounce?
-   should we handle length for input text?
