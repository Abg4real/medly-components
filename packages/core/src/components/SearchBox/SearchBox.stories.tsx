import { defaultTheme, SearchBoxTheme } from '@medly-components/theme/src';
import { Props } from './types';

export const optionsArray = [
    { value: 'Dummy1 option', label: 'Dummy1 option' },
    { value: 'Some 1', label: 'Dummy2 option' },
    { value: 'Dummy3 option', label: 'Dummy3 option' },
    { value: 'Dummy4 option', label: 'Dummy4 option' },
    { value: 'Dummy5 option', label: 'Dummy5 option' },
    { value: 'Dummy6 option', label: 'Dummy6 option' },
    { value: 'Dummy7 option', label: 'Dummy7 option' }
];

export const size: Props['size'][] = ['S', 'M'];

export const ThemeInterface: React.FC<SearchBoxTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.searchBox
};
