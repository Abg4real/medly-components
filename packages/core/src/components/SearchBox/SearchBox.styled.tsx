import { SvgIcon } from '@medly-components/icons';
import { defaultTheme } from '@medly-components/theme';
import { css, styled } from '@medly-components/utils';
import Options from '../SingleSelect/Options';
import { Props } from './types';

const getSearchBoxSize = ({ theme, searchBoxSize }: Props) => theme.searchBox.sizes[searchBoxSize].height;

const getIconWrapperStyle = ({ theme, searchBoxSize }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        height: 100%;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[searchBoxSize].iconSize};
        }
    `;
};

const getSmallOptionStyle = () => {
    return css`
        border-radius: 2rem 2rem 0 0;
        input {
            font-size: 1.4rem;
            line-height: 2.2rem;
            padding-left: 2rem;
        }
        ${Options.Style} {
            top: 3.8rem;
            border-radius: 0 0 2rem 2rem;
            li {
                span {
                    font-size: 1.4rem;
                    line-height: 2.2rem;
                }
            }
        }
        ${SvgIcon} {
            padding: 0.2rem;
        }
    `;
};

const getMediumOptionStyle = () => {
    return css`
        border-radius: 2.5rem 2.5rem 0 0;
        input {
            font-size: 1.6rem;
            line-height: 2.6rem;
            padding-left: 2.4rem;
        }
        ${Options.Style} {
            top: 4.6rem;
            border-radius: 0 0 2.5rem 2.5rem;
            li {
                span {
                    font-size: 1.6rem;
                    line-height: 2.6rem;
                }
            }
        }
        ${SvgIcon} {
            padding: 0.4rem;
        }
    `;
};

const activeOptionStyle = ({ theme }: Props) => {
    return css`
        border: solid 0.1rem ${theme.searchBox.active.borderColor};
        border-top: 0;
        box-shadow: 0 8px 16px rgba(96, 120, 144, 0.25);
        padding-bottom: 1.2rem;
        &::before {
            display: block;
            content: '';
            border-top: 1px solid ${theme.colors.grey[200]};
            width: 90%;
            height: 1px;
            margin: 0 auto 8px;
        }
    `;
};

const nonActiveOptionStyle = () => {
    return css`
        border: none;
        padding-bottom: 0;
        &::before {
            display: none;
        }
    `;
};

const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

const activeSearchBoxStyle = ({ theme }: Props) => {
    const {
        searchBox: { active }
    } = theme;
    return css`
        position: relative;
        box-shadow: 0 -0.8rem 1.6rem rgba(96, 120, 144, 0.25);
        border-bottom: 0 none;
        border-color: ${active.borderColor};
    `;
};

const nonActiveSearchBoxStyle = ({ theme }: Props) => {
    const {
        searchBox: { hover, active, default: defaultStyle }
    } = theme;

    return css`
        border-bottom: 0.1rem solid;
        border-color: ${defaultStyle.borderColor};
        border-radius: 4rem;
        &:focus-within {
            ${getBorderAndBoxShadow(active.borderColor, active.boxShadow)};
            ${SvgIcon} {
                * {
                    fill: ${active.borderColor};
                }
            }
        }
        &:hover {
            ${getBorderAndBoxShadow(hover.borderColor, hover.boxShadow)};
        }
        &:focus-within:hover {
            ${getBorderAndBoxShadow(active.borderColor, active.boxShadow)};
        }
    `;
};

export const SearchIconWrapper = styled('span')<Props & { isActive?: boolean }>`
    ${getIconWrapperStyle};
    ${SvgIcon} {
        * {
            fill: ${({ theme: { searchBox }, isActive }) => (isActive ? searchBox.active.iconColor : searchBox.default.iconColor)};
        }
        &:hover {
            background: ${({ theme }) => theme.searchBox.active.iconBg};
            border-radius: 2.5rem;
        }
    }
`;

export const CloseIconWrapper = styled('span')<Props & { isActive?: boolean }>`
    ${getIconWrapperStyle};
    ${SvgIcon} {
        * {
            fill: ${({
                theme: {
                    searchBox: { default: defaultTheme }
                }
            }) => defaultTheme.iconColor};
        }
        &:hover {
            background: ${({
                theme: {
                    searchBox: { default: defaultTheme }
                }
            }) => defaultTheme.iconBg};
            border-radius: 2.5rem;
            * {
                fill: ${({ theme }) => theme.colors.black};
            }
        }
    }
    &::after {
        content: '';
        width: 1px;
        height: ${({ searchBoxSize }) => (searchBoxSize === 'M' ? '3.2rem' : '2.4rem')};
        background: ${({ theme }) => theme.colors.grey[200]};
        display: ${({ isActive }) => (isActive ? 'block' : 'none')};
        margin-left: 0.8rem;
    }
`;

export const SearchBoxWrapper = styled('div')<Props & { isActive?: boolean }>`
    width: 25.6rem;
    display: flex;
    flex-direction: row;
    outline: none;
    box-sizing: border-box;
    height: ${getSearchBoxSize};
    border: 0.1rem solid;

    ${Options.Style} {
        box-shadow: none;
        border: none;
        box-sizing: content-box;
        left: -1px;
        ${({ isActive }) => (isActive ? activeOptionStyle : nonActiveOptionStyle)};

        li {
            padding: 0 2rem;
            &:last-child {
                border-radius: 0 0 1rem 1rem;
            }
        }
    }

    ${({ searchBoxSize }) => (searchBoxSize === 'M' ? getMediumOptionStyle : getSmallOptionStyle)};
    ${({ isActive }) => (isActive ? activeSearchBoxStyle : nonActiveSearchBoxStyle)};
`;

export const SearchBox = styled('input')<Props & { isActive?: boolean }>`
    background: transparent;
    color: ${({ theme, isActive }) => (isActive ? theme.colors.black : theme.searchBox.default.textColor)};
    padding: 0;
    outline: none;
    border: 0 none;
    width: 13.6rem;
    flex-grow: 1;

    &::placeholder {
        color: ${({ theme }) => theme.searchBox.default.placeholderTextColor};
    }
    &:focus {
        &::placeholder {
            color: ${({ theme }) => theme.searchBox.active.placeholderTextColor};
        }
    }
`;

SearchBox.defaultProps = {
    theme: defaultTheme
};
