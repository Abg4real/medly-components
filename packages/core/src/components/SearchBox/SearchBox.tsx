import { CloseIcon, SearchIcon } from '@medly-components/icons';
import { useCombinedRefs, WithStyle } from '@medly-components/utils';
import React, { FC, useCallback, useEffect, useRef, useState } from 'react';
import { getOptionsWithSelected } from '../SingleSelect/helpers';
import Options from '../SingleSelect/Options';
import { Option } from '../SingleSelect/types';
import { useKeyboardNavigation } from '../SingleSelect/useKeyboardNavigation';
import * as Styled from './SearchBox.styled';
import { Props } from './types';

export const SearchBox: FC<Props> & WithStyle = React.memo(
    React.forwardRef((props, ref) => {
        const { options: defaultOptions, searchBoxSize, placeholder, onSearchInputChange, onOptionSelected } = props;
        const inputRef = useCombinedRefs<HTMLInputElement>(ref, React.useRef(null)),
            isFocused = useRef(false),
            optionsRef = useRef<HTMLUListElement>(null),
            [isActive, setActive] = useState(false),
            [isTyping, updateIsTyping] = useState(false),
            [areOptionsVisible, showOptions] = useState(false),
            [options, setOptions] = useState(defaultOptions),
            [selectedOption, setSelectedOption] = useState({
                value: '',
                label: ''
            });

        useEffect(() => {
            setOptions(props.options);
            if (isTyping && props.options.length > 0) {
                setActive(true);
                showOptions(true);
                isFocused.current = true;
            } else {
                showOptions(false);
                setActive(false);
            }
        }, [props.options, isTyping]);

        const clearSearchText = useCallback(() => {
                inputRef.current.value = '';
                inputRef.current.focus();
                setActive(false);
                showOptions(false);
                updateIsTyping(false);
            }, []),
            handleChange = useCallback(
                (event: React.ChangeEvent<HTMLInputElement>) => {
                    const value = inputRef.current.value;
                    updateIsTyping(true);
                    onSearchInputChange(event.target.value);
                    if (value.length === 0) {
                        updateIsTyping(false);
                        isFocused.current = false;
                    }
                },
                [inputRef.current, options]
            ),
            handleOptionClick = useCallback(
                (option: Option) => {
                    if (!option.disabled && !Array.isArray(option.value) && onOptionSelected) {
                        inputRef.current.value = option.label;
                        showOptions(false);
                        onOptionSelected(option);
                    }
                    inputRef.current.focus();
                },
                [inputRef.current]
            ),
            selectOption = useCallback(
                (option: Option) => {
                    setOptions(getOptionsWithSelected(options, option));
                    setSelectedOption(option);
                },
                [options]
            ),
            showOptionsCB = useCallback(() => {
                showOptions(true);
                // @ts-ignore
                inputRef.current.setSelectionRange(inputRef.length, inputRef.length);
                inputRef.current.focus();
            }, [inputRef]),
            hideOptions = useCallback(() => {
                if (inputRef.current.value.length === 0) {
                    isFocused.current = false;
                }
            }, []);

        useKeyboardNavigation({
            isFocused,
            selectedOption,
            options,
            areOptionsVisible,
            selectOption,
            handleOptionClick,
            showOptions: showOptionsCB,
            optionsRef
        });

        return (
            <Styled.SearchBoxWrapper isActive={isActive} areOptionsVisible={areOptionsVisible} searchBoxSize={searchBoxSize}>
                <Styled.SearchBox
                    isActive={isActive}
                    placeholder={placeholder}
                    searchBoxSize={searchBoxSize}
                    onChange={handleChange}
                    onKeyDown={hideOptions}
                    onKeyUp={hideOptions}
                    ref={inputRef}
                />
                {areOptionsVisible && (
                    <Options
                        ref={optionsRef}
                        options={options}
                        highlightSelected={true}
                        variant="filled"
                        onOptionClick={handleOptionClick}
                    ></Options>
                )}
                <Styled.CloseIconWrapper isTyping={isTyping} searchBoxSize={searchBoxSize}>
                    {isTyping && <CloseIcon title="close icon" onClick={clearSearchText} />}
                </Styled.CloseIconWrapper>
                <Styled.SearchIconWrapper isActive={isActive} isTyping={isTyping} searchBoxSize={searchBoxSize}>
                    <SearchIcon title="search icon" />
                </Styled.SearchIconWrapper>
            </Styled.SearchBoxWrapper>
        );
    })
);

SearchBox.displayName = 'SearchBox';
SearchBox.Style = Styled.SearchBox;
SearchBox.defaultProps = {
    options: [],
    placeholder: 'Search',
    searchBoxSize: 'S'
};
