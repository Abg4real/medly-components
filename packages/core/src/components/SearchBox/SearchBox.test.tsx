import { fireEvent, getByPlaceholderText, render } from '@test-utils';
import React from 'react';
import { SearchBox } from './SearchBox';

describe('SearchBox', () => {
    it('should render placeholder text when specified', () => {
        const { container } = render(<SearchBox placeholder={'Search country'} />);
        const inputNode = getByPlaceholderText(container, 'Search country');
        expect(inputNode).toBeInTheDocument();
    });

    it('should render text box of specified size', () => {
        const { container } = render(<SearchBox />);
        expect(container).toMatchSnapshot();
    });

    it('should render search icon', () => {
        const { getByTitle } = render(<SearchBox />);
        expect(getByTitle('search icon')).toBeInTheDocument();
    });

    describe('close icon', () => {
        const props = {
            placeholder: 'search me',
            onSearchInputChange: jest.fn()
        };

        it('should render close icon when user starts typing', () => {
            const { container, getByTitle } = render(<SearchBox {...props} />),
                inputEl = getByPlaceholderText(container, 'search me') as HTMLInputElement;
            fireEvent.change(inputEl, { target: { value: 'R' } });
            expect(inputEl.value).toHaveLength(1);
            expect(getByTitle('close icon')).toBeInTheDocument();
        });

        it('should not render close icon in initial state', () => {
            const { queryByTitle } = render(<SearchBox {...props} />);
            expect(queryByTitle('close icon')).toBeFalsy();
        });

        it('should hide the close icon and maintain the focus on input when clicked on close icon', () => {
            const { container, getByTitle, queryByTitle } = render(<SearchBox {...props} />),
                inputEl = getByPlaceholderText(container, 'search me') as HTMLInputElement;
            fireEvent.change(inputEl, { target: { value: 'R' } });
            const closeIcon = getByTitle('close icon');
            expect(closeIcon).toBeInTheDocument();
            fireEvent.click(closeIcon);
            expect(inputEl.value).toHaveLength(0);
            expect(queryByTitle('close icon')).toBeNull();
            expect(document.activeElement === inputEl).toBeTruthy();
        });
    });

    describe('options', () => {
        const defaultReturnObj = { target: { value: 'Dummy' } };
        const props = {
            placeholder: 'search me',
            onSearchInputChange: jest.fn(),
            options: [
                { label: 'Dummy 1', value: 'Dummy 1' },
                { label: 'Dummy 2', value: 'Dummy 2' }
            ]
        };

        it('should not show options initially', () => {
            const { container } = render(<SearchBox {...props} />);

            expect(container.querySelector('ul')).toBeNull();
        });

        it('should render options when user search option specified', () => {
            const { container } = render(<SearchBox {...props} />),
                inputEl = getByPlaceholderText(container, 'search me') as HTMLInputElement;

            fireEvent.change(inputEl, defaultReturnObj);
            expect(container.querySelector('ul').children).toHaveLength(2);
            expect(container).toMatchSnapshot();
        });

        it('should call on OptionSelected when option is clicked', () => {
            const onOptionSelectedMock = jest.fn();
            const { container } = render(<SearchBox {...props} onOptionSelected={onOptionSelectedMock} />),
                inputEl = getByPlaceholderText(container, 'search me') as HTMLInputElement;
            fireEvent.change(inputEl, defaultReturnObj);
            fireEvent.click(container.querySelector('#Dummy-2'));
            expect(onOptionSelectedMock).toHaveBeenCalledWith({ value: 'Dummy 2', label: 'Dummy 2' });
        });
    });
});
