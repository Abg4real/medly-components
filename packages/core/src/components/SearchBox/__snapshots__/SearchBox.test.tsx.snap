// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox options should render options when user search option specified 1`] = `
.c13 {
  overflow: visible;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  font-size: 2.4rem;
  cursor: inherit;
}

.c13 * {
  fill-opacity: 1;
  fill: #607890;
}

.c10 {
  overflow: visible;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  font-size: 2.4rem;
  cursor: pointer;
}

.c10 * {
  fill-opacity: 1;
  fill: #607890;
}

.c7 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c5 {
  cursor: pointer;
  padding: 0 1.6rem;
  min-height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: #13181D;
  background-color: #ffffff;
}

.c5:hover {
  background-color: #eff2f4;
}

.c5 .c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c5 .c9 {
  margin-left: 1.6rem;
}

.c5 .c9 * {
  fill: #13181D;
}

.c3 {
  z-index: 1000;
  position: absolute;
  top: 5.6rem;
  left: 0;
  margin: 0;
  padding: 0;
  width: 100%;
  list-style: none;
  box-sizing: border-box;
  box-shadow: 0 2px 8px #B0BCC8;
  border-radius: 0 0 0.4rem 0.4rem;
}

.c3 li:first-child {
  border-radius: 0;
}

.c3 li:last-child {
  border-radius: 0 0 0.4rem 0.4rem;
}

.c16 .c9 {
  cursor: pointer;
  font-size: 2rem;
}

.c16 .c9 * {
  fill: #607890;
}

.c16 .c9:hover {
  background: transparent;
  border-radius: 2.5rem;
}

.c17 .c9 {
  cursor: pointer;
  font-size: 2rem;
}

.c17 .c9 * {
  fill: #607890;
}

.c17 .c9:hover {
  background: #e6effd;
  border-radius: 2.5rem;
}

.c12 {
  width: 4rem;
  min-width: 4rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c12 .c9 {
  cursor: pointer;
  font-size: 2rem;
}

.c12 .c9 * {
  fill: #005AEE;
}

.c12 .c9:hover {
  background: #e6effd;
  border-radius: 2.5rem;
}

.c15 .c9 {
  cursor: pointer;
  font-size: 2rem;
}

.c15 .c9 * {
  fill: #607890;
}

.c15 .c9:hover {
  background: #eff2f4;
  border-radius: 2.5rem;
}

.c15 .c9:hover * {
  fill: #303C48;
}

.c8 {
  width: 4rem;
  min-width: 4rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-right: 0.1rem solid #dfe4e9;
  margin: 0.6rem 0;
}

.c8 .c9 {
  cursor: pointer;
  font-size: 2rem;
}

.c8 .c9 * {
  fill: #607890;
}

.c8 .c9:hover {
  background: #eff2f4;
  border-radius: 2.5rem;
}

.c8 .c9:hover * {
  fill: #303C48;
}

.c14 .c2 {
  box-shadow: none;
  border: none;
  box-sizing: content-box;
  left: -1.5px;
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  border: none;
  padding-bottom: 0;
}

.c14 .c2::before {
  display: none;
}

.c14 .c2 svg {
  display: none;
}

.c14 .c2 strong {
  font-weight: normal;
}

.c14 .c2 .c4:hover {
  background-color: #F7F8F9;
}

.c14 .c2 {
  top: 3.6rem;
  border-radius: 0 0 2rem 2rem;
}

.c14 .c2 li {
  padding: 0 2rem;
}

.c14 .c2 li span,
.c14 .c2 li strong {
  font-size: 1.4rem;
  line-height: 2.2rem;
}

.c14 .c2::before {
  width: 82%;
}

.c14 .c9 {
  padding: 0.2rem;
}

.c14:focus-within .c11 .c9 * {
  fill: #005AEE;
}

.c0 {
  width: 25.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  outline: none;
  box-sizing: border-box;
  height: 4rem;
  border: 1.5px solid;
  position: relative;
  border-radius: 2rem 2rem 0 0;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
  border-color: #005AEE;
  border-width: 1.5px;
}

.c0 .c2 {
  box-shadow: none;
  border: none;
  box-sizing: content-box;
  left: -1.5px;
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  border: 1.5px solid #005AEE;
  border-top: 0;
  box-shadow: 0 0.6rem 0.8rem rgba(18,106,250,0.2);
  padding-bottom: 1.2rem;
}

.c0 .c2::before {
  display: block;
  content: '';
  border-top: 0.1rem solid #dfe4e9;
  height: 0.1rem;
  margin: 0 auto 0.8rem;
}

.c0 .c2 svg {
  display: none;
}

.c0 .c2 strong {
  font-weight: normal;
}

.c0 .c2 .c4:hover {
  background-color: #F7F8F9;
}

.c0 input {
  font-size: 1.4rem;
  line-height: 2.2rem;
  padding-left: 2rem;
}

.c0 .c2 {
  top: 3.6rem;
  border-radius: 0 0 2rem 2rem;
}

.c0 .c2 li {
  padding: 0 2rem;
}

.c0 .c2 li span,
.c0 .c2 li strong {
  font-size: 1.4rem;
  line-height: 2.2rem;
}

.c0 .c2::before {
  width: 82%;
}

.c0 .c9 {
  padding: 0.2rem;
}

.c1 {
  background: transparent;
  color: #13181D;
  padding: 0;
  outline: none;
  border: 0 none;
  width: 13rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-family: inherit;
  font-weight: 600;
}

.c1::-webkit-input-placeholder {
  color: #607890;
}

.c1::-moz-placeholder {
  color: #607890;
}

.c1:-ms-input-placeholder {
  color: #607890;
}

.c1::placeholder {
  color: #607890;
}

.c1:focus::-webkit-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::-moz-placeholder {
  color: #C7D0D8;
}

.c1:focus:-ms-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::placeholder {
  color: #C7D0D8;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="search me"
    />
    <ul
      class="c2 c3"
    >
      <li
        class="c4 c5"
        id="Dummy-1"
      >
        <span
          class="c6 c7"
        >
          Dummy 1
        </span>
      </li>
      <li
        class="c4 c5"
        id="Dummy-2"
      >
        <span
          class="c6 c7"
        >
          Dummy 2
        </span>
      </li>
    </ul>
    <span
      class="c8"
    >
      <svg
        class="c9 c10"
        fill="none"
        height="1em"
        title="close icon"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7A.996.996 0 1 0 5.7 7.11L10.59 12 5.7 16.89a.996.996 0 1 0 1.41 1.41L12 13.41l4.89 4.89a.996.996 0 1 0 1.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
          fill="#000"
          fill-opacity=".54"
        />
      </svg>
    </span>
    <span
      class="c11 c12"
    >
      <svg
        class="c9 c13"
        fill="none"
        height="1em"
        title="search icon"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M15.187 14.472h.79l4.24 4.26c.41.41.41 1.08 0 1.49-.41.41-1.08.41-1.49 0l-4.25-4.25v-.79l-.27-.28a6.5 6.5 0 0 1-5.34 1.48c-2.78-.47-5-2.79-5.34-5.59a6.505 6.505 0 0 1 7.27-7.27c2.8.34 5.12 2.56 5.59 5.34a6.5 6.5 0 0 1-1.48 5.34l.28.27zm-9.71-4.5c0 2.49 2.01 4.5 4.5 4.5s4.5-2.01 4.5-4.5-2.01-4.5-4.5-4.5-4.5 2.01-4.5 4.5z"
          fill="#000"
          fill-opacity=".54"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SearchBox should render text box of specified size 1`] = `
.c6 {
  overflow: visible;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  font-size: 2.4rem;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  fill: #607890;
}

.c4 {
  width: 4rem;
  min-width: 4rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 .c5 {
  cursor: pointer;
  font-size: 2rem;
}

.c4 .c5 * {
  fill: #607890;
}

.c4 .c5:hover {
  background: transparent;
  border-radius: 2.5rem;
}

.c2 {
  width: 4rem;
  min-width: 4rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-right: 0 none;
  margin: 0.6rem 0;
}

.c2 .c5 {
  cursor: pointer;
  font-size: 2rem;
}

.c2 .c5 * {
  fill: #607890;
}

.c2 .c5:hover {
  background: #eff2f4;
  border-radius: 2.5rem;
}

.c2 .c5:hover * {
  fill: #303C48;
}

.c0 {
  width: 25.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  outline: none;
  box-sizing: border-box;
  height: 4rem;
  border: 1.5px solid;
  position: relative;
  border-radius: 4rem;
  border-bottom: 1.5px solid;
  border-color: #98A7B7;
  border-radius: 4rem;
  -webkit-transition: border-color 100ms,box-shadow 100ms;
  transition: border-color 100ms,box-shadow 100ms;
}

.c0 .sc-AxhCb {
  box-shadow: none;
  border: none;
  box-sizing: content-box;
  left: -1.5px;
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  border: none;
  padding-bottom: 0;
}

.c0 .sc-AxhCb::before {
  display: none;
}

.c0 .sc-AxhCb svg {
  display: none;
}

.c0 .sc-AxhCb strong {
  font-weight: normal;
}

.c0 .sc-AxhCb .sc-AxiKw:hover {
  background-color: #F7F8F9;
}

.c0 input {
  font-size: 1.4rem;
  line-height: 2.2rem;
  padding-left: 2rem;
}

.c0 .sc-AxhCb {
  top: 3.6rem;
  border-radius: 0 0 2rem 2rem;
}

.c0 .sc-AxhCb li {
  padding: 0 2rem;
}

.c0 .sc-AxhCb li span,
.c0 .sc-AxhCb li strong {
  font-size: 1.4rem;
  line-height: 2.2rem;
}

.c0 .sc-AxhCb::before {
  width: 82%;
}

.c0 .c5 {
  padding: 0.2rem;
}

.c0:focus-within {
  border-color: #005AEE;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
}

.c0:focus-within .c3 .c5 * {
  fill: #005AEE;
}

.c0:hover {
  border-color: #607890;
  box-shadow: 0 0.2rem 0.8rem rgba(96,120,144,0.2);
}

.c0:focus-within:hover {
  border-color: #005AEE;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
}

.c1 {
  background: transparent;
  color: #13181D;
  padding: 0;
  outline: none;
  border: 0 none;
  width: 13rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-family: inherit;
  font-weight: 600;
}

.c1::-webkit-input-placeholder {
  color: #607890;
}

.c1::-moz-placeholder {
  color: #607890;
}

.c1:-ms-input-placeholder {
  color: #607890;
}

.c1::placeholder {
  color: #607890;
}

.c1:focus::-webkit-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::-moz-placeholder {
  color: #C7D0D8;
}

.c1:focus:-ms-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::placeholder {
  color: #C7D0D8;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="Search"
    />
    <span
      class="c2"
    />
    <span
      class="c3 c4"
    >
      <svg
        class="c5 c6"
        fill="none"
        height="1em"
        title="search icon"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M15.187 14.472h.79l4.24 4.26c.41.41.41 1.08 0 1.49-.41.41-1.08.41-1.49 0l-4.25-4.25v-.79l-.27-.28a6.5 6.5 0 0 1-5.34 1.48c-2.78-.47-5-2.79-5.34-5.59a6.505 6.505 0 0 1 7.27-7.27c2.8.34 5.12 2.56 5.59 5.34a6.5 6.5 0 0 1-1.48 5.34l.28.27zm-9.71-4.5c0 2.49 2.01 4.5 4.5 4.5s4.5-2.01 4.5-4.5-2.01-4.5-4.5-4.5-4.5 2.01-4.5 4.5z"
          fill="#000"
          fill-opacity=".54"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;
